### DISCUSSION TITLE
Shorten root path.

### DISCUSSION THREAD

**Discussion by madelaney** [nvim-tree/nvim-tree.lua#1921] [General]
It is possible to change the root path to a shortened format? For example if my full path is `~/work/gitlab.mldelaney.io/project_a/example-repo`, rather then displaying the full path could we display `~/w/g/p/example-repo`?

**Comment by gegoune** [comment]
Look at `:h nvim-tree.renderer.root_folder_label`.

**Comment by madelaney** [comment]
Thanks @gegoune , I was able to write ( poorly ) a function that did what I was looking for.

Here is the solution I came up with (I'm going to assume there are optimizations that can be done):
```lua
function shorten_root_path(value)
  local home = os.getenv("HOME")
  local path = ""
  local sep  = "/"

  local t = {}
  for str in string.gmatch(value, "([^" .. sep .. "]+)") do
    table.insert(t, str)
  end

  local size = 0
  for _ in pairs(t) do size = size + 1 end

  local index = 0

  if value:find(home, 1, true) == 1 then
    local _, count = string.gsub(home, "/", "")
    index = count + 1
    path = "~"
  end

  for k, v in pairs(t) do
    if k >= index then
      if (path:len() == 0) then
        path = string.sub(v, 1, 1)
      elseif (v:len() == 1) then
        path = path .. sep .. v
      elseif (k == size) then
        path = path .. sep .. v
      else
        path = path .. sep .. string.sub(v, 1, 1)
      end
    end
  end

  return path
end
```

**Reply by alex-courtis** [reply]
It looks like there might be something built into vim to do this. Take a look at the tab titles in a clean setup:

![20230117_112026](https://user-images.githubusercontent.com/1505378/212783912-625f8ac6-b1b1-439e-877b-33ea3880cbe9.png)

Not sure how they are set...

**Comment by litoj** [comment]
@madelaney I think this is something you're looking for:
```lua
root_folder_label = function(path)
  path = path:gsub(os.getenv("HOME"), "~", 1)
  return path:gsub("([a-zA-Z])[a-z]+", "%1") .. path:gsub(".*[^a-zA-Z].?", "")
end,
```

turns `~/Documents/work/projects-fit/packages/profit-theses-backend`
into `~/D/w/p-f/p/p-t-backend`

I honestly think this should become the standard.

**Reply by alex-courtis** [reply]
> I honestly think this should become the standard.

If we were building nvim-tree today, we might do that. Unfortunately we don't like to break existing behaviour.
